scipy/stats/tests/test_stats.py

```diff
@@ -2412,21 +2345,66 @@ def test_gh16952(self):
         assert_array_equal(res.mode, [1, 1, 1, 1])
         assert_array_equal(res.count, [2, 3, 3, 3])
 
-
-def test_mode_futurewarning():
-    a = [1, 2, 5, 3, 5]
-
-    future_msg = "Unlike other reduction functions..."
-    with pytest.warns(FutureWarning, match=future_msg):
-        res = stats.mode(a)
-    assert_array_equal(res, ([5], [2]))
-
-    # no FutureWarning if `keepdims` is specified
-    res = stats.mode(a, keepdims=True)
-    assert_array_equal(res, ([5], [2]))
-
-    res = stats.mode(a, keepdims=False)
-    assert_array_equal(res, [5, 2])
+    def test_gh9955(self):
+        # The behavior of mode with empty slices (whether the input was empty
+        # or all elements were omitted) was inconsistent. Test that this is
+        # resolved: the mode of an empty slice is NaN and the count is zero.
+        res = stats.mode([])
+        ref = (np.nan, 0)
+        assert_equal(res, ref)
+
+        res = stats.mode([np.nan], nan_policy='omit')
+        assert_equal(res, ref)
+
+        a = [[10., 20., 20.], [np.nan, np.nan, np.nan]]
+        res = stats.mode(a, axis=1, nan_policy='omit')
+        ref = ([20, np.nan], [2, 0])
+        assert_equal(res, ref)
+
+        res = stats.mode(a, axis=1, nan_policy='propagate')
+        ref = ([20, np.nan], [2, 3])
+        assert_equal(res, ref)
+
+        z = np.array([[], []])
+        res = stats.mode(z, axis=1)
+        ref = ([np.nan, np.nan], [0, 0])
+        assert_equal(res, ref)
+
+    @pytest.mark.filterwarnings('ignore::RuntimeWarning')  # np.mean warns
+    def test_gh9955b(self):
+        # A more thorough test of gh-9955 with empty arrays based on
+        # `test_axis_nan_policy.test_empty`
+
+        def small_sample_generator(n_dims):
+            # return all possible "small" arrays in up to n_dim dimensions
+            for i in n_dims:
+                # "small" means with size along dimension either 0 or 1
+                for combo in combinations_with_replacement([0, 1, 2], i):
+                    yield np.zeros(combo)
+
+        for sample in small_sample_generator([2, 3]):
+            # this test is only for arrays of zero size
+            if not sample.size == 0:
+                continue
+
+            # need to test for all valid values of `axis` parameter, too
+            for axis in range(-sample.ndim, sample.ndim):
+                res = stats.mode(sample, axis=axis)
+
+                ref = np.mean(sample, axis=axis) * np.nan  # mode is always NaN
+                assert_equal(res.mode, ref)
+                assert np.shape(res.mode) == ref.shape
+
+                ref[np.isnan(ref)] = 0  # count is always zero
+                assert_equal(res.count, ref)
+                assert res.count.shape == ref.shape
+
```

Comment:
We don't necessarily need to include this test, but I wanted to make sure it passed.